### PR TITLE
MediaPlayerElement now support fullscreen mode

### PR DESCRIPTION
**Pull request by baptiste-nv**
Issue: #
None

## PR Type
What kind of change does this PR introduce?
Feature

## What is the current behavior?
MediaPlayerElement does not support fullscreen mode

## What is the new behavior?
MediaPlayerElement now support fullscreen mode

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tested code with current [supported SDKs](../README.md#supported)
- [X] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [X] Contains **NO** breaking changes

### CODE REVIEW

File: src/Uno.UI.Wasm/WasmScripts/Uno.UI.js [hunk 1]
```diff
@@ -823,41 +823,31 @@ var Uno;
         (function (Interop) {
             class ManagedObject {
                 static init() {
-                    if (Uno.UI.WindowManager.isHosted) {
-                        console.debug("Hosted Mode: skipping IndexDB initialization");
-                    }
-                    else {
+                    if (!ManagedObject.assembly) {
```
Comment (davidjohnoliver) [inline src/Uno.UI.Wasm/WasmScripts/Uno.UI.js]: Remove this file.

File: src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs [hunk 2]
```diff
@@ -118,7 +117,68 @@ private static void OnAutoPlayChanged(DependencyObject sender, DependencyPropert
...
+			});
+		}
+
+		private object _mainContent;
+
+		private void ToogleFullScreen(bool showFullScreen)
+		{
+			if (showFullScreen)
+			{
+				Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs]: Why are you dispatching this?

File: src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs [hunk 3]
```diff
@@ -118,7 +117,68 @@ private static void OnAutoPlayChanged(DependencyObject sender, DependencyPropert
...
+				{
+					ApplicationView.GetForCurrentView().TryEnterFullScreenMode();
+				});
+
+				_mainContent = (Windows.UI.Xaml.Window.Current.Content as Frame).Content;
+				(Windows.UI.Xaml.Window.Current.Content as Frame).Content = _layoutRoot;
+
+#if __ANDROID__
+				this.RemoveView(_layoutRoot);
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs]: Shouldn't you call this first? I'm surprised it works.

File: src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs [hunk 4]
```diff
@@ -118,7 +117,68 @@ private static void OnAutoPlayChanged(DependencyObject sender, DependencyPropert
...
+		private void ToogleFullScreen(bool showFullScreen)
+		{
+			if (showFullScreen)
+			{
+				Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
+				{
+					ApplicationView.GetForCurrentView().TryEnterFullScreenMode();
+				});
+
+				_mainContent = (Windows.UI.Xaml.Window.Current.Content as Frame).Content;
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs]: This is a little brittle:
1. There's no requirement for the Window.Content to be a Frame
2. If app code tries to access the visual tree while the media player is in full screen, it's going to get a surprise.
3. If there's a page navigation while the media player is full screen, the new page is effectively going to be erased when it leaves fullscreen (and the navigation stack will probably be screwed up).
etc

Did you look into doing it closer to how UWP does? Ie using FullScreenMediaRoot or whatever it's called. The implementation would probably be similar to that for Popup.

File: src/Uno.UI.Wasm/WasmScripts/Uno.UI.js (hunk 1 above)
Comment (baptiste-nv) [reply src/Uno.UI.Wasm/WasmScripts/Uno.UI.js]: Checked with @jeromelaban, no effect, we can keep it

File: src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs (hunk 2 above)
Comment (baptiste-nv) [reply src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs]: Underlying native methods need to run on UI thread

File: src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs (hunk 3 above)
Comment (baptiste-nv) [reply src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs]: It works. But I'll check that, it's maybe clearer

File: src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs (hunk 4 above)
Comment (baptiste-nv) [reply src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs]: I'll check that

File: src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs [hunk 5]
```diff
@@ -159,29 +219,21 @@ private void OnMediaFailed(Windows.Media.Playback.MediaPlayer session, object ar
...
-				}
+				TogglePosterImage(false);
 			});
 		}
 
-		#endregion
+#endregion
 
-		#region AreTransportControlsEnabled Property
+#region AreTransportControlsEnabled Property
```
Comment (MatFillion) [inline src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs]: why remove the tabs on all #regions ?

File: src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs (hunk 3 above)
Comment (baptiste-nv) [reply src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs]: Fixed

File: src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs (hunk 2 above)
Comment (baptiste-nv) [reply src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs]: Right! fixed

File: src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs (hunk 4 above)
Comment (baptiste-nv) [reply src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs]: @davidjohnoliver Updated. Could you review again please?

File: src/Uno.UI/UI/Xaml/Window.Android.cs [hunk 6]
```diff
@@ -26,8 +30,29 @@ public Window()
 
 		private void InternalSetContent(UIElement value)
 		{
-			_content = value;
-			ApplicationActivity.Instance?.SetContentView(value);
+			if (_main == null)
```
Comment (davidjohnoliver) [inline src/Uno.UI/UI/Xaml/Window.Android.cs]: I prefer this approach, but did you test what happens when you restart the activity? (Ie hit back button to back out of the app, and restore it by pressing the icon.) I think you may need to adjust ApplicationActivity.InitializeComponent().

Comment (baptiste-nv) [reply src/Uno.UI/UI/Xaml/Window.Android.cs]: You are right, updated

File: src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs (hunk 3 above)
Comment (baptiste-nv) [reply src/Uno.UI/UI/Xaml/Controls/MediaPlayer/MediaPlayerElement.cs]: Done

**Pull request merged**